sql-plugin/src/main/scala/org/apache/spark/sql/rapids/stringFunctions.scala

```diff
@@ -772,6 +772,20 @@ class GpuRLikeMeta(
 case class GpuRLike(left: Expression, right: Expression)
   extends GpuBinaryExpression with ImplicitCastInputTypes with NullIntolerant  {
 
+  private val cudfPattern = right.asInstanceOf[GpuLiteral].value match {
```

Comment:
Should this be lazy so we don't try to evaluate it every time this node is copied in the plan or the node is otherwise wielded in a way where we're not trying to execute it (e.g.: on the driver)?